dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -143,4 +161,42 @@ public void getIconData(HttpServletResponse response, @PathVariable String iconK
     webMessage.setMessage(String.format("Icon %s deleted", iconKey));
     return webMessage;
   }
+
+  private void downloadDefaultIcon(String defaultIcon, HttpServletResponse response)
+      throws IOException, NotFoundException {
+    Resource icon = iconService.getDefaultIconResource(defaultIcon);
+
+    response.setHeader("Cache-Control", CacheControl.maxAge(TTL, TimeUnit.DAYS).getHeaderValue());
+    response.setContentType(MediaType.SVG_UTF_8.toString());
```

Comment:
`downloadDefaultIcon` is used in `"/{iconKey}/icon.svg"` and `"/{uid}/icon"`. In the latter we also return custom icons. I just read about `png` format being used for custom icons in the issue description. Is it ok for users that the custom icons will be an png while default icons will be a svg in `"/{uid}/icon"`? I know we set the `content-type`. Its more about the usability of this feature.

Reply:
@stian-sandvold

Reply:
It's a good question @teleivo and I was hoping to merge this and leave the responsibility of interpreting the type of image to the client. png is just one supported type for custom icons; So there will always be some need to figure out the type.

The idea here was to serve _any_ icon through the same endpoint as well, so I think this is a natural problem clients will have to deal with to be able to fetch anything from the same endpoint.




I'm happy to hear any suggestions on alternative solutions :D Also, I am hoping to have this merged soon, so we can have android and web start working on supporting this new endpoint in their apps (If this is how we end up doing it)